### PR TITLE
PR: Add dialog to confirm downloaded installer installation and update related code organization (Application)

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

See https://github.com/spyder-ide/spyder/issues/19112#issuecomment-1228678444

A preview showing the new dialog without having the installer downloaded:

installer_download

A preview showing the new dialog when the installer was already downloaded previously:

installer

### Issue(s) Resolved

Part of #19112

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

**Comment by dalthviz** [comment]
Note: Seems like the MacOS installer workflow is failing due to the release of `black` 2.10.0

**Comment by mrclary** [comment]
> Note: Seems like the MacOS installer workflow is failing due to the release of `black` 2.10.0

Yep, looks like we should pin that to `<=22.8.0`.

**Review by mrclary — APPROVED** [review]
LGTM 👍🏼

File: spyder/workers/updates.py [hunk 1]
```diff
@@ -142,3 +148,86 @@ def start(self):
...
                 pass
+
+
+class WorkerDownloadInstaller(QObject):
+    """
+    Worker that donwloads standalone installers for Windows
+    and MacOS without blocking the Spyder user interface.
+    """
+    sig_ready = Signal(str)
+    sig_download_progress = Signal(int, int)
```
Comment (ccordoba12) [inline spyder/workers/updates.py:159]: Please add docstrings for these two signals.

File: spyder/workers/updates.py [hunk 2]
```diff
@@ -142,3 +148,86 @@ def start(self):
...
+        logger.debug("Downloading installer executable")
+        tmpdir = tempfile.gettempdir()
+        is_full_installer = (is_module_installed('numpy') or
+                             is_module_installed('pandas'))
+        if os.name == 'nt':
+            name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
+                                                else 'lite')
+        else:
+            name = 'Spyder{}.dmg'.format('' if is_full_installer
+                                            else '-Lite')
```
Comment (ccordoba12) [inline spyder/workers/updates.py:189]:
```suggestion
            name = 'Spyder{}.dmg'.format('' if is_full_installer else '-Lite')
```

File: spyder/workers/updates.py [hunk 3]
```diff
@@ -142,3 +148,86 @@ def start(self):
...
+            dir_path, self.latest_release_version)
+        os.makedirs(installer_dir_path, exist_ok=True)
+        for file in os.listdir(dir_path):
+            if file not in [__version__, self.latest_release_version]:
+                remove = osp.join(dir_path, file)
+                os.remove(remove)
+
+        installer_path = osp.join(installer_dir_path, name)
+        self.installer_path = installer_path
+        if (not osp.isfile(installer_path)):
```
Comment (ccordoba12) [inline spyder/workers/updates.py:205]:
```suggestion
        if not osp.isfile(installer_path):
```

File: spyder/workers/updates.py [hunk 4]
```diff
@@ -142,3 +148,86 @@ def start(self):
...
+            self._download_installer()
+        except UpdateDownloadCancelledException:
+            if self.installer_path:
+               os.remove(self.installer_path)
+            return
+        except HTTPError:
+            error_msg = _('Unable to retrieve installer information.')
+        except URLError:
+            error_msg = _('Unable to connect to the internet. <br><br>Make '
+                          'sure the connection is working properly.')
```
Comment (ccordoba12) [inline spyder/workers/updates.py:226]:
```suggestion
            error_msg = _('Unable to connect to the internet. <br><br>'
                          'Make sure the connection is working properly.')
```

File: spyder/workers/updates.py [hunk 5]
```diff
@@ -142,3 +148,86 @@ def start(self):
...
+        except Exception:
+            error_msg = _('Unable to download installer.')
```
Comment (ccordoba12) [inline spyder/workers/updates.py:228]:
```suggestion
            error_msg = _('Unable to download the installer.')
```

**Review by ccordoba12 — COMMENTED** [review]
This looks pretty nice! Thanks a lot @dalthviz for your work on it.

File: spyder/plugins/application/widgets/install.py [hunk 6]
```diff
@@ -117,9 +117,9 @@ class UpdateInstallerDialog(QDialog):
     Parameters
     ----------
     current_value: int
-        Number of bits downloaded until now.
+        Size of the data downloaded until now.
     total: int
-        Total expected number of bits to be downloaded.
+        Total size file expected to be downloaded.
```
Comment (ccordoba12) [inline spyder/plugins/application/widgets/install.py:122]:
```suggestion
        Total size of the file expected to be downloaded.
```

File: spyder/workers/updates.py [hunk 7]
```diff
@@ -155,8 +155,28 @@ class WorkerDownloadInstaller(QObject):
...
     sig_download_progress = Signal(int, int)
+    """
+    Signal to get the download progress.
+
+    Parameters
+    ----------
+    current_value: int
+        Size of the data downloaded until now.
+    total: int
+        Total size file expected to be downloaded.
```
Comment (ccordoba12) [inline spyder/workers/updates.py:178]:
```suggestion
        Total size of the file expected to be downloaded.
```

**Review by ccordoba12 — COMMENTED** [review]
@dalthviz, last small suggestions for you, then this should be ready.

**Pull request merged**